src/components/Reports/GoalCalculator/SharedComponents/GoalCalculatorGrid/GoalCalculatorGrid.tsx

```diff
@@ -1,196 +1,387 @@
-    setFieldValue('gridData', updatedData);
+  const handleDelete = (id: string | number) => {
+    const rowId = id.toString();
+
+    deleteSubBudgetCategory({
+      variables: {
+        input: {
+          accountListId,
+          id: rowId,
+        },
+      },
+      optimisticResponse: {
+        deleteSubBudgetCategory: {
+          __typename: 'SubBudgetCategoryDeleteMutationPayload',
+          id: rowId,
+        },
+      },
+      update: (cache, { data }) => {
+        if (data?.deleteSubBudgetCategory?.id) {
+          cache.updateFragment<UpdateSubBudgetCategoriesFragment>(
+            {
+              id: `PrimaryBudgetCategory:${category.id}`,
+              fragment: UpdateSubBudgetCategoriesFragmentDoc,
+              fragmentName: 'UpdateSubBudgetCategories',
+            },
+            (data) => {
+              if (!data) {
+                return data;
+              }
+              return {
+                ...data,
+                id: data.id,
+                subBudgetCategories: (data.subBudgetCategories || []).filter(
+                  (cat) => cat.id !== rowId,
+                ),
+              };
+            },
+          );
+        }
+      },
+    });
   };
 
   const processRowUpdate = (newRow: GridValidRowModel) => {
+    // Don't process the total row - it's read-only and calculated
     if (newRow.id === 'total') {
       return newRow;
     }
 
-    const updatedData = values.gridData.map((item) =>
-      item.id === newRow.id
-        ? {
-            ...item,
-            name: newRow.name as string,
-            amount: newRow.amount as number,
-          }
-        : item,
-    );
-    setFieldValue('gridData', updatedData);
+    // Extract the updated values from the grid row
+    const rowId = newRow.id as string;
+    const label = newRow.label as string;
+    const amount = newRow.amount as number;
+
+    try {
+      // Validate the updated row data using yup schema
+      subBudgetCategorySchema.validateSync({ label, amount });
+
+      // Clear any existing validation errors for this row since validation passed
+      setCellErrors((prev) => {
+        const updated = { ...prev };
+        delete updated[`${rowId}-label`];
+        delete updated[`${rowId}-amount`];
+        return updated;
+      });
+
+      updateSubBudgetCategory({
+        variables: {
+          input: {
+            accountListId,
+            attributes: {
+              id: rowId,
+              label,
+              amount,
+            },
+          },
+        },
+        optimisticResponse: {
+          updateSubBudgetCategory: {
+            __typename: 'SubBudgetCategoryUpdateMutationPayload',
+            subBudgetCategory: {
+              __typename: 'SubBudgetCategory',
+              id: rowId,
+              label,
+              amount,
+              // Preserve the original category value from the server data
+              category:
```

Comment:
This works. But since the category never gets updated, during a mutation, you could also change the update mutation definition to be this:

```gql
mutation UpdateSubBudgetCategory(
  $input: SubBudgetCategoryUpdateMutationInput!
) {
  updateSubBudgetCategory(input: $input) {
    subBudgetCategory {
      # Omit the category instead of using the fragment
      id
      label
      amount
    }
  }
}
```